orca-kotlin/src/main/kotlin/com/netflix/spinnaker/orca/ext/Stage.kt

```diff
@@ -103,34 +103,47 @@ fun Stage.allAfterStagesSuccessful(): Boolean =
 fun Stage.anyBeforeStagesFailed(): Boolean =
   beforeStages().any { it.status in listOf(TERMINAL, STOPPED, CANCELED) }
 
-fun Stage.anyAfterStagesFailed(): Boolean =
-  afterStages().any { it.status in listOf(TERMINAL, STOPPED, CANCELED) }
-
 fun Stage.allAfterStagesComplete(): Boolean =
   afterStages().all { it.status.isComplete }
 
 fun Stage.hasTasks(): Boolean =
   tasks.isNotEmpty()
 
-fun Stage.hasAfterStages(): Boolean =
-  firstAfterStages().isNotEmpty()
-
 inline fun <reified O> Stage.mapTo(pointer: String): O = mapTo(pointer, O::class.java)
 
 inline fun <reified O> Stage.mapTo(): O = mapTo(O::class.java)
 
-fun Stage.shouldFailPipeline(): Boolean =
-  context["failPipeline"] in listOf(null, true)
+fun Stage.failureStatus(default: ExecutionStatus = TERMINAL): ExecutionStatus {
+  // Because not all stages pass the full context to their child
+  // If this is a synthetic stage we must look at our parent to see if continuePipeline or failPipeline
+  // have been specified.
+  fun getFromParent(param: String, default: Boolean): Boolean {
+    var stage: Stage? = this
+
+    while (stage != null) {
+      if (stage.context[param] != null) {
+        return (stage.context[param] == true)
+      }
+
+      stage = if (stage.syntheticStageOwner != null) stage.parent() else null
+    }
 
-fun Stage.shouldContinueOnFailure(): Boolean =
-  context["continuePipeline"] == true
+    return default
+  }
+
+  // true if explicitly set to true
```

Comment:
Not sure this inline comment is necessary here?

Reply:
yeah, i wanted to make clear why one is defaulted to `true` and one to `false`, but I renamed the `default` param and I think it's clearer now and no longer needs the comment